tests/console/kdump_and_crash.pm

```diff
@@ -46,10 +46,18 @@ sub run {
     # all but PPC64LE arch's vmlinux images are gzipped
     my $suffix = get_var('OFW') ? '' : '.gz';
     assert_script_run 'find /var/crash/';
+
     my $crash_cmd = "echo exit | crash `ls -1t /var/crash/*/vmcore | head -n1` /boot/vmlinux-`uname -r`$suffix";
-    assert_script_run "$crash_cmd", 600;
-    validate_script_output "$crash_cmd", sub { m/PANIC/ }, 600;
+    script_run("$crash_cmd |& tee /dev/$serialdev; echo crash-status-\$? > /dev/$serialdev", 0);
+    if (wait_serial(qr/PANIC:\s([^\s]+)(.*?\n){1,7}.*crash-status-0/, 600)) {
+        return;
+    }
+    else {
+        record_soft_failure 'boo#1090659 - crash: invalid kernel virtual address';
```

Comment:
Hm, looks a bit generic because now this branch would always be hit in case the above wait_serial either did not find PANIC, non-zero exit code or timeout.
You could check the actual return of wait_serial in detail